activerecord/lib/arel/visitors/to_sql.rb

```diff
@@ -18,7 +18,24 @@ def compile(node, collector = Arel::Collectors::SQLString.new)
         accept(node, collector).value
       end
 
+      def accept(object, collector)
+        visit object, collector
+      end
+
       private
+        def visit(object, collector)
```

Comment:
Maybe worth a comment describing why this is here?

Especially because, as a perf change, there's no test to enforce it -- so this could easily go away in a later refactoring otherwise.

Reply:
We only use ToSQL visitors now, optional collector (splat args) is only for depth_first and dot visitors which are not used for us.
So I've changed to apply this change to base Visitor directly.
https://github.com/rails/rails/pull/36483/commits/85b4ba2836437ff30ff565005b2500500200611b#diff-13eb804616e021fefb0ba1e129089296R27